tutorials/text/deduplication/semantic/semantic_e2e.ipynb

```diff
@@ -0,0 +1,843 @@
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>2</th>\n",
+       "      <td>TinyStories-train.txt-2108502</td>\n",
+       "      <td>TinyStories-train.txt-578142</td>\n",
+       "      <td>1.000000</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>3</th>\n",
+       "      <td>TinyStories-train.txt-649801</td>\n",
+       "      <td>TinyStories-train.txt-2036969</td>\n",
+       "      <td>0.863873</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>4</th>\n",
+       "      <td>TinyStories-train.txt-899672</td>\n",
+       "      <td>TinyStories-train.txt-649801</td>\n",
+       "      <td>1.000000</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "                              id                         max_id  \\\n",
+       "0  TinyStories-train.txt-2036969  TinyStories-train.txt-2036969   \n",
+       "1   TinyStories-train.txt-578142  TinyStories-train.txt-2036969   \n",
+       "2  TinyStories-train.txt-2108502   TinyStories-train.txt-578142   \n",
+       "3   TinyStories-train.txt-649801  TinyStories-train.txt-2036969   \n",
+       "4   TinyStories-train.txt-899672   TinyStories-train.txt-649801   \n",
+       "\n",
+       "   cosine_sim_score  \n",
+       "0          0.000000  \n",
+       "1          0.832831  \n",
+       "2          1.000000  \n",
+       "3          0.863873  \n",
+       "4          1.000000  "
+      ]
+     },
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "pairwise_path = os.path.join(cache_path, \"semantic_dedup\", \"pairwise_results\")\n",
+    "\n",
+    "pd.read_parquet(os.path.join(pairwise_path, \"cluster_0.parquet\")).head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b5bd81da",
+   "metadata": {},
+   "source": [
+    "##### Visualizing Similarity in our dataset\n",
+    "\n",
+    "Depending on our dataset size we can read through all of the files and plot how much data is similar to one another.\n",
+    "Here we show how to read file by file and then perform a reduce. \n",
+    "\n",
+    "In our dataset we can see that ~20% of our data has cosine_similarity of 0.9 or more.\n",
+    "\n",
+    "Based on the analysis here we can decide what our `eps` should be. \n",
+    "However in this tutorial we pre-ran with eps set and perform_removal to be True.\n",
+    "\n",
+    "However ideally, users do this analysis, inspect the duplicates, come up with an `eps` and then run a pipeline that includes the `IdentifyDuplicates` stage.\n",
+    "\n",
+    "And finally perform removal.\n",
+    "\n",
+    "**NOTE : If you run with `use_id_generator=False` (which affects removal performance at large scale) you will see the actual ids of instead of the int ids. You can then also inspect the original dataset for those ids, and that can also help your decision for `eps`**"
```

Comment:
I think this is a great section for users to know how to interpret the results and play around with some parameters here. Could we flesh it out more?